game/plugins/src/main/kotlin/gg/rsmod/plugins/content/items/amuletofglory/amulet_of_glory.plugin.kts

```diff
@@ -0,0 +1,66 @@
+package gg.rsmod.plugins.content.items.amuletofglory
+
+import gg.rsmod.plugins.content.magic.TeleportType
+import gg.rsmod.plugins.content.magic.canTeleport
+import gg.rsmod.plugins.content.magic.teleport
+
+val GLORY = intArrayOf(
+        Items.AMULET_OF_GLORY1, Items.AMULET_OF_GLORY2, Items.AMULET_OF_GLORY3,
+        Items.AMULET_OF_GLORY4, Items.AMULET_OF_GLORY5, Items.AMULET_OF_GLORY6
+)
+
+private val SOUNDAREA_ID = 200
+private val SOUNDAREA_RADIUS = 5
+private val SOUNDAREA_VOLUME = 1
+
+private val LOCATIONS = mapOf(
+        "Edgeville" to Tile(3086, 3500, 0),
+        "Karamja" to Tile(2917, 3176, 0),
+        "Draynor Village" to Tile(3103, 3249, 0),
+        "Al kharid" to Tile(3292, 3165, 0)
+)
+
+GLORY.forEach { glory ->
+    LOCATIONS.forEach { location, tile ->
+        on_equipment_option(glory, option = location) {
+            player.queue(TaskPriority.STRONG) {
+                player.teleport(tile)
+            }
+        }
+    }
+}
+
+
+fun Player.teleport(endTile : Tile) {
+    if (canTeleport(TeleportType.GLORY)) {
+        if (hasEquipped(EquipmentType.AMULET, *GLORY)) {
+            world.spawn(AreaSound(tile, SOUNDAREA_ID, SOUNDAREA_RADIUS, SOUNDAREA_VOLUME))
+            equipment[EquipmentType.AMULET.id] = set()
+            message(message())
+            teleport(endTile, TeleportType.GLORY)
+        }
+    }
+}
+
+fun Player.set(): Item ? {
```

Comment:
Now that this returns an item, the function name should be more descriptive of what it does. Self-documenting code is what you should aim for

for example `fun Player.getAmuletReplacement`